javalib/src/main/scala/java/util/Hashtable.scala

```diff
@@ -14,69 +14,65 @@ package java.util
 
 import java.{util => ju}
 
-import scala.collection.mutable
-
-import ScalaOps._
-
-class Hashtable[K, V] private (inner: mutable.HashMap[Box[Any], V])
+class Hashtable[K, V] private (inner: ju.HashMap[K, V])
     extends ju.Dictionary[K,V] with ju.Map[K, V] with Cloneable with Serializable {
 
   def this() =
-    this(mutable.HashMap.empty[Box[Any], V])
+    this(new ju.HashMap[K, V]())
 
-  def this(initialCapacity: Int) = this()
+  def this(initialCapacity: Int) =
+    this(new ju.HashMap[K, V](initialCapacity))
 
-  def this(initialCapacity: Int, loadFactor: Float) = this()
+  def this(initialCapacity: Int, loadFactor: Float) =
+    this(new ju.HashMap[K, V](initialCapacity, loadFactor))
 
-  def this(t: ju.Map[_ <: K, _ <: V]) = {
-    this()
-    putAll(t)
-  }
+  def this(t: ju.Map[_ <: K, _ <: V]) =
+    this(new ju.HashMap[K, V](t)) // this lets `null` through instead of throwing NPEs
```

Comment:
> On the JVM that would throw an NPE, but we don't care too much about NPEs so we let them pass through in this case. That seemed better than doing another loop for the sake of checking the nulls.

I think it is important to keep the behavior consistent with other methods. Otherwise very strange stuff can happen. E.g. with the current implementation:

```scala
if (m.containsKey(null)) {
  m.get(null) // throws NPE
}
```

Reply:
I see your point, but it's going to be really annoying to do fully. In particular that means that we'll have to implement a dedicated entry iterator and hence entry set, so that their `setValue` refuses `null`s. It seems overkill.

I'd rather allow `null`s everywhere, if you really want consistency.

Reply:
I have updated the code to allow `null`s everywhere.